File: Server/user-service/src/main/java/com/randima/userservice/model/Transaction.java
package com.randima.userservice.model;

import lombok.Data;

import java.time.LocalDate;

@Data
public class Transaction {
    private Integer id;
    private Integer bookId;
    private Integer universityId;
    private LocalDate checkin_date;
    private String checkout_date;
    private Integer renew_flag;
    private Integer fine;
    private boolean isReturned;
    private LocalDate returnedDate;
}

File: Client/library-management-system/src/app/model/Transaction.ts
import { User } from './User';
import { Book } from './Book';

export interface Transaction{
    id : number,
    bookId : number,
	userId : number,
	checkin_date : Date,
	checkout_date : Date,
	renew_flag : number,
	fine : number,
	returned : boolean,
	user:User,
	book: Book
}